doc/developer/material.md

```diff
@@ -371,6 +371,245 @@ void TriangleMeshComponent::handleMeshLoading( const Ra::Asset::GeometryData* da
+ 
+### Interface proposal
+As glsl allows pre-declaration of functions, we are able to define the main interfaces even 
+if one could find dependencies between them when implementing a particular instance. one exemple of such dependencies 
+is the Blinn-Phong BSDF that could take parameters either from the "Material" object in Radium or from color attributes 
+of a vertex.
+
+#### BSDF interface
+Implementing a BSDF in Radium requires 2 steps :
+1. Implement the C++ interface `Ra::Engine::Material` 
+2. Implement the glsl interface described below.
+
+For implementing the C++ interface, refer to the Radium documentation. One thing to remember is that the 
+method `Material::getMaterialName()` must return a string that contains the `name_of_the_BSDF` implemented
+in a file named `name_of_the_BSDF.glsl` that must be preloaded in a `glNamedString` to allow inclusion by others.
+
+This file must only contains the implementation of the BSDF interface, with no `void main(){...}` nor access to
+vertex attribs or so on ...
+
+This file must contain an inclusion guard :
+```glsl
+#ifndef METALLICROUGHNESS_GLSL
+#define METALLICROUGHNESS_GLSL
+.
+.
+.
+#endif
+```
+This file must define the following
+
+```glsl
+    /// Concrete definition of the Material structure that contains the BSDF parameters
+    struct Material {
+        ...
+    };
+
+    /// Returns the base color, or albedo, of the material, at the surface coordinates defined by texCoord.
+    /// The returned color will eventually be used to infer if the fragment is transparent or not.
+    /// The alpha channel could then vary from 0 (totally transparent) to 1 (totally opaque)
+    vec4 getBaseColor(Material material, vec2 texCoord);
+
+    /// Returns the so called "Diffuse Color" of the material, at the surface coordinates defined by texCoord.
+    /// This could be the same that the base color or obtained by a more or less complex computation (Fresnel, ...)
+    /// The alpha channel is the same than the one computed for getBaseColor().
+    vec4 getDiffuseColor(Material material, vec2 texCoord);
+    
+    /// Returns the so called "Specular Color" of the material, at the surface coordinates defined by texCoord.
+    /// This could be the same that the base color (without the alpha channel) or obtained by 
+    /// a more or less complex computation
+    vec3 getSpecularColor(Material material, vec2 texCoord);
+    
+    /// Return the bsdf value for the material, at surface coordinates defined by texCoord,
+    /// for the incoming and outgoing directions `wi` and `wo`. These directions MUIST be in local frame.
+    /// The local Frame is the Frame wher the Geometric normal is the Z axis, 
+    /// and the tangent defined the X axis.
+    vec3 evaluateBSDF(Material material, vec2 texCoord, vec3 wi, vec3 wo);
+
+```
+    
+#### Microgeometry interface
+Defining the micro-geometry programmatically allows to decorelates the geometric sampling from the appearance parameters 
```

Comment:
Do you mean as texture ? I'm not sure this is the point of this documentation.
We may refer the reader to any computer graphics introduction.

Reply:
I mean either as textures or procedures. This is specified and references to a practical introduction (learnopengl.com) and more indepth introduction (Real-time rendering) to CG is added.